packages/db/src/definitions/networkGenealogies.ts

```diff
@@ -0,0 +1,124 @@
+import gql from "graphql-tag";
+
+import {Definition} from "./types";
+
+export const networkGenealogies: Definition<"networkGenealogies"> = {
+  createIndexes: [],
+  idFields: ["ancestor", "descendant"],
+  typeDefs: gql`
+    type CandidateSearchResult {
+      network: Network!
+      alreadyTried: [ID]! #will include all networks returned
+    }
+
+    type NetworkGenealogy implements Resource {
+      id: ID!
+      ancestor: Network
+      descendant: Network
+    }
+
+    input NetworkGenealogyInput {
+      ancestor: ResourceReferenceInput!
+      descendant: ResourceReferenceInput!
+    }
+
+    extend type Network {
+      possibleAncestors(
+        alreadyTried: [ID]!
+        limit: Int # will default to 5
+      ): [CandidateSearchResult]!
+      possibleDescendants(
+        alreadyTried: [ID]!
+        limit: Int # will default to 5
+      ): [CandidateSearchResult]!
+    }
+  `,
+  resolvers: {
+    NetworkGenealogy: {
+      ancestor: {
+        resolve: async ({ancestor}, __, {workspace}) => {
+          const result = await workspace.get("networks", ancestor.id);
+          return result;
+        }
+      },
+      descendant: {
+        resolve: async ({descendant}, __, {workspace}) =>
+          await workspace.get("networks", descendant.id)
+      }
+    },
+    Network: {
+      possibleAncestors: {
+        resolve: async ({id}, {limit, alreadyTried}, {workspace}) => {
+          const network = await workspace.get("networks", id);
+          const queryLimit = limit ? limit : 5;
+          const result = await workspace.find("networks", {
+            selector: {
+              "historicBlock.height": {
+                $gte: null,
```

Comment:
Oddly, to get the sort to work I had to add the `$gte: null` line and also explicitly exclude the network I'm checking with. I'm a bit concerned about the `$ne` because that is done in-memory according to the docs which may be a hit to performance.

From the [pouchdb docs](https://pouchdb.com/guides/mango-queries.html): "For instance, $regex, $ne, and $not cannot use on-disk indexes, and must use in-memory filtering instead"